mtp_cashbook/apps/disbursements/forms.py

```diff
@@ -244,15 +246,31 @@ class RecipientBankAccountForm(DisbursementForm):
         help_text=_('For example, 10-20-30'),
         validators=[validate_sort_code],
     )
+    roll_number = forms.CharField(
+        required=False,
+        label=_('Building society roll number (if applicable)'),
+        help_text=_('Up to 18 characters, which can include A to Z, 0 to 9, space, / and -'),
+    )
 
     def clean(self):
         cleaned_data = super().clean()
-        if roll_number_required(
-            cleaned_data.get('sort_code'), cleaned_data.get('account_number')
-        ):
+        sort_code = cleaned_data.get('sort_code')
+        account_number = cleaned_data.get('account_number')
+        if roll_number_required(sort_code, account_number):
```

Comment:
at this point, sort code and account number could already be invalid. i don't think this check should run if they're already invalid…